libnopegl/src/hmap.c

```diff
@@ -118,17 +118,17 @@ static struct hmap_ref ref_from_entry(const struct hmap *hm, const struct hmap_e
         return NO_REF;
     struct hmap_ref ref = {
         .bucket_id = entry->bucket_id,
-        .entry_id  = (int)(entry - hm->buckets[entry->bucket_id].entries),
+        .entry_id  = (size_t)(entry - hm->buckets[entry->bucket_id].entries),
     };
     return ref;
 }
 
 static struct hmap_entry *fixed_entry_from_ref(const struct hmap *hm, struct hmap_ref ref,
-                                               struct hmap_ref removed, int current_id)
+                                               struct hmap_ref removed, size_t current_id)
 {
     if (!HAS_REF(ref))
         return NULL;
-    const int need_fix =
+    const size_t need_fix =
```

Comment:
It looks like need_fix is a boolean. Isn't better to cast it later on ? or simplify use a `if` statement and sub explictely by `1`.

Reply:
Yep, done with a cast